sdk/src/trace/tracer.cc

```diff
@@ -36,11 +38,16 @@ namespace sdk
 {
 namespace trace
 {
+const std::shared_ptr<opentelemetry::trace::NoopTracer> Tracer::kNoopTracer =
+    std::make_shared<opentelemetry::trace::NoopTracer>();
```

Comment:
Why a shared_ptr ?

This memory will never be reclaimed.

This should be enough:

```
opentelemetry::trace::NoopTracer kNoopTracer();
```

Reply:
I was trying to avoid making multiple copies of NoopTracer, but looking at it again, shared_ptr might actually have more overhead due to ref counts.

Reply:
I think I remember the main reason for why I put this as `shared_ptr`.

The [NoopTracer](https://github.com/open-telemetry/opentelemetry-cpp/blob/150256c1720b279a1dfc7c380d17e13e90cee15d/api/include/opentelemetry/trace/noop.h#L94) inherits from `std::enable_shared_from_this<NoopTracer>`. 

And IIUC, this means an exception would be thrown if no shared_ptr exists at the time of usage. I just removed the shared_ptr and the tests started failing with `std::bad_weak_ptr` exception.

I'm not super familiar with `enable_shared_from_this`, so please let me know if my understanding is incorrect.